### ISSUE TITLE
Does the builtin comment feature miss the `gco` etc function?

### ISSUE THREAD

**Issue by XXiaoA** [neovim/neovim#28979]
### Problem

Thanks to the great devotion from developers ,we got a nice builtin comment feature in 0.10 version,.
But, having used `Comment.nvim` for a pretty long time,  I'm addict to the `gco` function, which add a new comment in the next line. e.g.:
```
something (cursor here then press gco)
```
```
something
// cursor here
```
Though I've asked  echasnovski in reddit ([here](https://www.reddit.com/r/neovim/comments/1bwlvrt/comment/ky9owl6/)], I still want to make a struggle to discuss the need for this feature. Apparently, we can press o to insert the comment then press gcc. But I think this method breaks the undo history and we need to  hit the keyboard more times.

### Expected behavior

add the `gco` `gcO` keymaps

**Comment by seblj** [comment]
Something like `nmap gco oo<Esc>gccfoa<BS>` is a very naive but easy way you can map it yourself

**Comment by towry** [comment]
`:h :undoj`

**Comment by XXiaoA** [comment]
Well, thanks for the advice. I came up with this mapping
`nmap("gco", "oo<ESC>:normal gcc<CR>A<BS>")`
It approximately meet my need, except the blink for adding and delete a temporary char. I would appreciate it if you have any better idea.

**Comment by XXiaoA** [comment]
I replaced the `o` with "hair space" (Hex 200a), then it look better

**Comment by rockyzhang24** [comment]
I extracted and simplify the relevent code from [Comment.nvim](https://github.com/numToStr/Comment.nvim). Not tested it too much but it should work. Give it a shot.

```lua
local function comment_above_or_below(lnum)
  local row, _ = unpack(vim.api.nvim_win_get_cursor(0))
  local comment_row = row + lnum
  local l_cms, r_cms = string.match(vim.bo.commentstring, '(.*)%%s(.*)')
  l_cms = vim.trim(l_cms)
  r_cms = vim.trim(r_cms)
  if r_cms ~= 0 then
    r_cms = ' ' .. r_cms
  end
  vim.api.nvim_buf_set_lines(0, comment_row, comment_row, false, { l_cms .. ' ' .. r_cms})
  vim.api.nvim_win_set_cursor(0, { comment_row + 1, 0 })
  vim.api.nvim_command('normal! ==')
  vim.api.nvim_win_set_cursor(0, { comment_row + 1, #vim.api.nvim_get_current_line() - #r_cms - 1 })
  vim.api.nvim_feedkeys('a', 'ni', true)
end

vim.keymap.set('n', "gco", function()
  comment_above_or_below(0)
end)

vim.keymap.set('n', "gcO", function()
  comment_above_or_below(-1)
end)
```

**Comment by XXiaoA** [comment]
> Give it a shot.

Thanks a lot! I tried it and find there is the logic of adding space is sort of beyond my expectation. So I modified it based on yours to meet my own need.
```lua
local function comment_above_or_below(lnum)
    local row, _ = unpack(vim.api.nvim_win_get_cursor(0))
    local comment_row = row + lnum
    local l_cms, r_cms = string.match(vim.bo.commentstring, "(.*)%%s(.*)")
    vim.api.nvim_buf_set_lines(0, comment_row, comment_row, false, { l_cms .. r_cms })
    vim.api.nvim_win_set_cursor(0, { comment_row + 1, 0 })
    vim.api.nvim_command("normal! ==")
    vim.api.nvim_win_set_cursor(0, { comment_row + 1, #vim.api.nvim_get_current_line() - #r_cms - 1 })
    vim.api.nvim_feedkeys("a", "ni", true)
end
```

(Maybe this issue could be closed?)

**Comment by rockyzhang24** [comment]
@XXiaoA After your modification, the newly added comment line in html will be `<!--x-->` instead of `<!-- x -->` where `x` represents the cursor. If this is what you expect, ignore me here.

**Comment by XXiaoA** [comment]
@rockyzhang24 Yeah, that's what I exactly want, follow the native commentstring. Your version would add the redundant space for `-- %s` like `-- | ` (| is cursor, note there is a space after it.) I don't want deal with it, so I delete it. Another reason: I can easy add space for `<!--x-->` by the autopairs plugin

**Comment by rockyzhang24** [comment]
My bad. I omitted `#` in `r_cms` empty checking. Fixed it.
